ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ethernet_interfaces.schema.yml

```diff
@@ -0,0 +1,809 @@
+# yaml-language-server: $schema=../../../../plugins/plugin_utils/schema/avd_meta_schema.json
+# Line above is used by RedHat's YAML Schema vscode extension
+# Use Ctrl + Space to get suggestions for every field. Autocomplete will pop up after typing 2 letters.
+type: dict
+keys:
+  ethernet_interfaces:
+    type: list
+    primary_key: name
+    convert_types:
+    - dict
+    items:
+      type: dict
+      keys:
+        name:
+          type: str
+          required: true
+        description:
+          type: str
+        shutdown:
+          type: bool
+        speed:
+          type: str
+          description: Speed can be interface_speed or forced interface_speed or auto interface_speed
+        mtu:
+          type: int
+          convert_types:
+          - str
+        l2_mtu:
+          type: int
+          convert_types:
+          - str
+          description: |
+            "l2_mtu" should only be defined for platforms supporting the "l2 mtu" CLI
+        vlans:
+          type: str
+          convert_types:
+          - int
+          description: |
+            List of switchport vlans as string.
+            For a trunk port this would be a range like "1-200,300"
+            For an access port this would be a single vlan "123"
+        native_vlan:
+          type: int
+          convert_types:
+          - str
+        native_vlan_tag:
+          type: bool
+          description: If setting both native_vlan and native_vlan_tag, native_vlan_tag takes precedence
+        mode:
+          type: str
+          valid_values: ["access", "dot1q-tunnel", "trunk", "trunk phone"]
+        phone:
+          type: dict
+          keys:
+            trunk:
+              type: str
+              valid_values: ["tagged", "untagged"]
+            vlan:
+              type: int
+              min: 1
+              max: 4094
+              convert_types:
+              - str
+        l2_protocol:
+          type: dict
+          keys:
+            encapsulation_dot1q_vlan:
+              type: int
+              convert_types:
+              - str
+              description: Vlan tag to configure on sub-interface
+        trunk_groups:
+          type: list
+          items:
+            type: str
+        type:
+          type: str
+          valid_values: ["routed", "switched", "l3dot1q", "l2dot1q"]
+          description: l3dot1q and l2dot1q are used for sub-interfaces. The parent interface should be defined as routed.
+        snmp_trap_link_change:
+          type: bool
+        flowcontrol:
+          type: dict
+          keys:
+            received:
+              type: str
+              valid_values: ["desired", "on", "off"]
```

Comment:
So this used to be this in documentation:
```
    flowcontrol:
      received: < "received" | "send" | "on" >
```
Looking at a device it should look like:
```
Leaf1(config-if-Et8)#flowcontrol receive ?
  desired  Allow but do not require flow-control capable link partner
  off      Forbid flow-control capable link partner
  on       Require flow-control capable link partner

Leaf1(config-if-Et8)#flowcontrol send ?
  desired  Allow but do not require flow-control capable link partner
  off      Forbid flow-control capable link partner
  on       Require flow-control capable link partner
```

Though the template is only doing this:

```
   48 {%         if ethernet_interface.flowcontrol.received is arista.avd.defined %}
   49    flowcontrol receive {{ ethernet_interface.flowcontrol.received }}
   50 {%         endif %}
```

so this is a documentation fix at the same time - maybe worth fixing it in the documentation?

Reply:
Yes, I will fix!

Reply:
fix in commit [ab0b4ed](https://github.com/aristanetworks/ansible-avd/pull/2107/commits/ab0b4edbc9d0ce8af93bce7ef1c746b6695487c8)